src/EditorFeatures/CSharpTest/RemoveUnusedParametersAndValues/RemoveUnusedValueAssignmentTests.cs

```diff
@@ -7239,5 +7239,67 @@ string M(object obj)
     }
 }", options: PreferDiscard, parseOptions: new CSharpParseOptions(LanguageVersion.CSharp8));
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedValues)]
+        [WorkItem(39344, "https://github.com/dotnet/roslyn/issues/39344")]
+        public async Task AssignmentInTry_UsedInFinally_NoDiagnostic()
+        {
+            await TestDiagnosticMissingAsync(
+@"using System;
```

Comment:
Do we expect to report if it's not used in the finally? 

```c# 
void M(int i)
{
    bool b = false;
    try 
    { 
        if (i == 0) b = true;
    }
    finally 
    {
    }
}
```

Reply:
Yes, that works prior to the fix - both assignments are highlighted as redundant. I can still add a unit test.